workflows/pipe-common/scripts/autoscale_sge.py

```diff
@@ -291,6 +354,28 @@ def delete_host(self, host, queue=_MAIN_Q, hostgroup=_ALL_HOSTS, skip_on_failure
         self._remove_host_from_host_group(host, hostgroup, skip_on_failure=skip_on_failure)
         self._remove_host_from_grid_engine(host, skip_on_failure=skip_on_failure)
 
+    def get_resource_demand(self, expired_jobs):
+        avaliable_slots = 0
+        demand_slots = 0
+        for line in self.cmd_executor.execute_to_lines(GridEngine._SHOW_EXECUTION_HOSTS_SLOTS):
```

Comment:
Probably the number of available slots calculation can be extracted to a private method.